platforms/core-runtime/build-configuration/src/main/java/org/gradle/internal/buildconfiguration/DaemonJvmPropertiesConfigurator.java

```diff
@@ -53,32 +55,38 @@ public void execute(ProjectInternal project) {
-                    .zip(task.getJvmVersion().zip(task.getJvmVendor().orElse("any"), Pair::of),
+                    .zip(task.getLanguageVersion().zip(task.getVendor().orElse(DefaultJvmVendorSpec.any()), Pair::of),
                         (platforms, versionVendor) -> {
-                            String vendor = versionVendor.getRight();
+                            JvmVendorSpec vendor = versionVendor.getRight();
                             JavaToolchainSpec toolchainSpec = project.getObjects().newInstance(DefaultToolchainSpec.class);
                             toolchainSpec.getLanguageVersion().set(versionVendor.getLeft());
-                            if (!vendor.equals("any")) {
-                                toolchainSpec.getVendor().set(JvmVendorSpec.of(vendor));
+                            if (!vendor.equals(DefaultJvmVendorSpec.any())) {
+                                toolchainSpec.getVendor().set(vendor);
                             }
-                            JavaToolchainResolverService resolverService = project.getServices().get(JavaToolchainResolverService.class);
-                            if (!platforms.isEmpty() && !resolverService.hasConfiguredToolchainRepositories()) {
-                                // TODO revisit failure condition, but don't require generation as long as its results are not used
-//                                throw new UnconfiguredToolchainRepositoriesResolver();
+                            if (platforms.isEmpty()) {
                                 return emptyMap();
                             }
+
+                            JavaToolchainResolverService resolverService = project.getServices().get(JavaToolchainResolverService.class);
+                            if (!resolverService.hasConfiguredToolchainRepositories()) {
+                                throw new UnconfiguredToolchainRepositoriesResolver();
+                            }
                             Map<BuildPlatform, Optional<URI>> buildPlatformOptionalUriMap = platforms.stream()
                                 .collect(Collectors.toMap(platform -> platform,
                                     platform -> resolverService.tryResolve(new DefaultJavaToolchainRequest(toolchainSpec, platform)).map(JavaToolchainDownload::getUri)));
-                            return buildPlatformOptionalUriMap.entrySet().stream()
+                            Map<BuildPlatform, URI> platformToDownloadUri = buildPlatformOptionalUriMap.entrySet().stream()
                                 .filter(e -> e.getValue().isPresent())
                                 .collect(Collectors.toMap(Map.Entry::getKey, e -> e.getValue().get()));
+                            if (platformToDownloadUri.isEmpty()) {
+                                throw new IllegalStateException("Toolchain resolvers did not return download URLs providing a JDK matching " + toolchainSpec + " for any of the requested platforms " + platforms);
```

Comment:
Since this is a new message, the IDE should incorporate this on the error handling for the Daemon toolchain to offer quick fixes. I'll create a task to do it: 

<img width="1181" alt="Screenshot 2025-02-05 at 13 15 32" src="https://github.com/user-attachments/assets/8abc72f3-6391-4541-b0de-47bfb7d35c42" />